boards/arm/colibri_imx7d_m4/doc/colibri_imx7d_m4.rst

```diff
@@ -0,0 +1,339 @@
++============+=======================+========================+=======================+======================+
+| DDR        | 0x80000000-0xFFFFFFFF | 0x80000000-0xDFFFFFFF  | 0x10000000-0x1FFEFFFF | 2048MB (less for M4) |
++------------+-----------------------+------------------------+-----------------------+----------------------+
+| OCRAM_PXP  | 0x00940000-0x00947FFF | 0x20240000-0x20247FFF  | 0x00940000-0x00947FFF | 32KB                 |
++------------+-----------------------+------------------------+-----------------------+----------------------+
+| OCRAM_EPDC | 0x00920000-0x0093FFFF | 0x20220000-0x2023FFFF  | 0x00920000-0x0093FFFF | 128KB                |
++------------+-----------------------+------------------------+-----------------------+----------------------+
+| OCRAM      | 0x00900000-0x0091FFFF | 0x20200000-0x2021FFFF  | 0x00900000-0x0091FFFF | 128KB                |
++------------+-----------------------+------------------------+-----------------------+----------------------+
+| TCMU       | 0x00800000-0x00807FFF | 0x20000000-0x20007FFF  |                       | 32KB                 |
++------------+-----------------------+------------------------+-----------------------+----------------------+
+| TCML       | 0x007F8000-0x007FFFFF |                        | 0x1FFF8000-0x1FFFFFFF | 32KB                 |
++------------+-----------------------+------------------------+-----------------------+----------------------+
+| OCRAM_S    | 0x00180000-0x00187FFF | 0x20180000-0x20187FFF  | 0x00000000-0x00007FFF | 32KB                 |
++------------+-----------------------+------------------------+-----------------------+----------------------+
+| QSPI Flash |                       |                        | 0x08000000-0x0BFFFFFF | 64MB                 |
++------------+-----------------------+------------------------+-----------------------+----------------------+
+
+
+References
+==========
+
+- `i.MX 7 Dual Reference Manual`_ from page 190 (section 2.1.2 and 2.1.3)
+- `Toradex Wiki`_
+
+
+At compilation time you have to choose which RAM will be used. This configuration is
+done in the file ``boards/arm/colibri_imx7d_m4/colibri_imx7d_m4.dts`` with "zephyr,flash" (when CONFIG_XIP=y) and "zephyr,sram"
+properties. The available configurations are:
+
+"zephyr,flash"
+- &ddr_code
+- &tcml_code
+- &ocram_code
+- &ocram_s_code
+- &ocram_pxp_code
+- &ocram_epdc_code
+
+"zephyr,sram"
+- &ddr_sys
+- &tcmu_sys
+- &ocram_sys
+- &ocram_s_sys
+- &ocram_pxp_sys
+- &ocram_epdc_sys
+
+
+Below you will find the instructions to load and run Zephyr on M4 from A7 using u-boot.
+
+Copy the compiled zephyr.bin to the first EXT partition of the SD card and plug into the
+board. Power it up and stop the u-boot execution.
+Set the u-boot environment variables and run the zephyr.bin from the appropriated memory
+configured in the Zephyr compilation:
+
+.. code-block:: console
+  setenv bootm4 'ext2load mmc 0:1 $m4addr $m4fw && dcache flush && bootaux $m4addr'
```

Comment:
need a blank line after the ``.. code-block`` directive and then all the lines within the code block need to be indented to the "c" in "code-block"